### DISCUSSION TITLE
[Feature] Guardar Partida

### DISCUSSION THREAD

**Discussion by emrodk** [Godot-en-Espanol/Node-City#6] [Ideas]
- Piensan que se debe guardar la partida?
- Qué guardarían?
 - la última partida en progreso y que se recupere el juego?
 - solo el score y el stage de la última partida ganada?
 - múltiples slots para partidas?
 - tiempo de juego?
 - 
 -

**Comment by azagaya** [comment]
Creo que para definir que se debería guardar necesitamos avanzar un poco más en las mecánicas. Quizás si tenes bufs que pasan de una partida a otra, deberías poder guardarlas. 
En principio, no guardaría un estado intermedio de una partida. Solo el nivel al que se llego, y los bufs o cosas así que tenga el personaje. No porque sea un problema, sino para evitar el tema de save&try a cada paso. Aunque igual eso ya sería problema del jugador.
Multiples slots es siempre una buena idea.
Tiempo de juego tmb, y además es algo que no cuesta agregar.

**Comment by jjmontes** [comment]
Si definimos que hay niveles  de complejidad creciente, en el que se puede jugar y terminar el juego (por decir algo, hay 20 niveles) creo que la opción de guardar partida puede servir. Si es onda un nivel atrás del otro sin ningún propósito salvo matar el tiempo, no le veo sentido.

Si se puede guardar, creo que sería bueno tener diferentes slots (para diferentes personas compartiendo el dispositivo), pero creo que la implementaría con autoguardado y recuperación automática, y al perder, volvés a comenzar.

Dependiendo de las cosas existentes en el juego, se definiría que es lo que se guarda. Me parece que falta definir cosas antes la posibilidad de guardar y recuperar.

**Comment by Gabo-Salcedo** [comment]
Repitiendo un poco lo que dicen, considero que deberíamos definir primero si van a ver niveles... de otra forma no le veo sentido guardar la partida... a menos que estemos hablando del constructor de niveles... eso si se debería guardar, ¿En tal caso se puede compartir un nivel construido para que otra persona lo juegue????

**Reply by azagaya** [reply]
Diria que si. Deberian poder compartirse.

**Comment by gfazzito** [comment]
Comparto la opinión de definirlo más adelante (cuando estén las mecánicas). Igual dejo una idea para debatirlo después: si llega a ser muy complicado y largo el nivel podríamos incluir un guardado rápido dentro de la partida.

**Comment by FROSADO** [comment]
Yo veo el guardar partida algo más como, suspendo el juego porque tengo que irme y volver por donde lo dejé. 
* Guardaría solo el estado actual de la partida, nivel / powerups / vida ...
* Guardado automático tras cada paso de nivel 
*  No más de la partida en curso. De forma que al inicio aparece un menu de nueva partida o continuar si existe dato de guardado.

**Reply by azagaya** [reply]
Cuando decís "solo el estado actual de la partida" te referís a que si dejo un nivel a la mitad, puedo volver a ese nivel?
